packages/payments-service/src/payment-request/controller.ts

```diff
@@ -0,0 +1,17 @@
+import { Headers, Controller, Post, Param, Body } from '@nestjs/common'
+import { PaymentRequestDto, PaymentRequestHeaders, PaymentRequestParams } from './dto'
+import { PaymentRequestProvider } from './provider'
+
+@Controller('request')
+export class PaymentRequestController {
+  constructor(private readonly paymentRequestProvider: PaymentRequestProvider) {}
+
+  @Post('/:paymentId')
+  async createRequest(
+    @Headers() paymentRequestHeaders: PaymentRequestHeaders,
+    @Param() { paymentId }: PaymentRequestParams,
+    @Body() paymentRequest: PaymentRequestDto,
+  ): Promise<void> {
+    this.paymentRequestProvider.create(paymentRequestHeaders, paymentRequest, paymentId)
```

Comment:
```suggestion
    await this.paymentRequestProvider.create(paymentRequestHeaders, paymentRequest, paymentId)
```

Does this need to be awaited? Not returned this time so presuming you want to await it